src/core/lib/lattice/hal/default/dcrtpoly.cpp

```diff
@@ -2564,25 +2564,28 @@ void DCRTPolyImpl<VecType>::FastBaseConvqToBskMontgomery(
     }
 
     // mod mtilde = 2^16
-    std::vector<uint16_t> result_mtilde(n);
+    uint64_t mtilde      = (uint64_t)1 << 16;
+    uint64_t mtilde_half = mtilde >> 1;
+
+    std::vector<uint64_t> result_mtilde(n);
```

Comment:
the vector should be constructed with the default value of 0 for its elements:
std::vector<uint64_t> result_mtilde(n, 0);
and the line 2573 -> "result_mtilde[k] = 0;" should be removed. This will improve the performance

Reply:
Fixed as suggested.